### PR TITLE
installer script for EMR

### PR DESCRIPTION
**Pull request by pvam**

### CODE REVIEW

File: rubix_admin/installer.py [hunk 1]
```diff
@@ -18,13 +19,12 @@ def setup_parsers(cls, sub_parser):
...
         cls.install_parser.add_argument("-r", "--rpm", required=True,
                                         help="Path to RPM file")
-        cls.install_parser.add_argument("-a", "--rpm-args", default="",
+        cls.install_parser.add_argument("-a", "--rpm-args", default="--ignoreos",
                                         help="Arguments to rpm command")
         cls.install_parser.set_defaults(func=cls.install_cmd)
 
     @classmethod
     def install_cmd(cls, args):
-        logging.info("Installing %s" % args.rpm)
```
Comment (vrajat) [inline rubix_admin/installer.py]: Reason for removing this log ? If it is too verbose, please move it to debug level.

File: rubix_admin/installer.py [hunk 2]
```diff
@@ -50,7 +50,15 @@ def _scp(cls, args):
...
-                     os.path.join(args.config["remote_packages_path"],
-                                  os.path.basename(args.rpm))))
+        sudo('rpm -U %s %s' %
+             (args.rpm_args,
+              os.path.join(args.config["remote_packages_path"],
+                           os.path.basename(args.rpm))))
+
+        sudo("cp -a /usr/lib/rubix/lib/* /usr/lib/presto/lib/plugin/hive-hadoop2/")
+        sudo("cp -a /usr/lib/rubix/lib/* /usr/lib/hadoop/lib/")
+        sudo("mkdir -p /media/ephemeral0")
```
Comment (vrajat) [inline rubix_admin/installer.py]: why is `/media/ephemeral0` required ? What file are created there ?
If it is logs, then these should go into `/var/logs/rubix` and you should change rubix to accept the log location as a parameter.

Comment (pvam) [inline rubix_admin/installer.py]: No logs, raw cached files are stored in that directory.

Comment (pvam) [inline rubix_admin/installer.py]: Can be specified through 'hadoop.cache.data.dirprefix.list' hadoop property, defaults to '/media/ephemeral0'.

Comment (pvam) [inline rubix_admin/installer.py]: I will see if passing it as a -D parameter works or will find some other way.

Comment (vrajat) [inline rubix_admin/installer.py]: OK. MySQL etc. store in `/var/lib/mysql`* . 
The question is if `/` points to a small root partition ? Before taking a final decision, can you paste the o/p of `mount` to check the default mount points in a emr node ?  

* https://stackoverflow.com/questions/2659782/where-is-the-actual-data-in-a-mysql-db-stored-on-a-linux-machine

Comment (pvam) [inline rubix_admin/installer.py]: EMR nodes come with 10gb default space on root partition. Can be changed during the cluster creation though. 

Mount o/p:
proc on /proc type proc (rw,relatime)
sysfs on /sys type sysfs (rw,relatime)
devtmpfs on /dev type devtmpfs (rw,relatime,size=7672784k,nr_inodes=1918196,mode=755)
devpts on /dev/pts type devpts (rw,relatime,gid=5,mode=620,ptmxmode=000)
tmpfs on /dev/shm type tmpfs (rw,relatime)
/dev/xvda1 on / type ext4 (rw,noatime,data=ordered)
none on /proc/sys/fs/binfmt_misc type binfmt_misc (rw,relatime)
/dev/xvdb1 on /emr type xfs (rw,relatime,attr2,inode64,noquota)
/dev/xvdb2 on /mnt type xfs (rw,relatime,attr2,inode64,noquota)
/dev/xvdc on /mnt1 type xfs (rw,relatime,attr2,inode64,noquota)
/dev/xvdd on /mnt2 type xfs (rw,relatime,attr2,inode64,noquota)
/dev/xvde on /mnt3 type xfs (rw,relatime,attr2,inode64,noquota)
cgroup on /cgroup/blkio type cgroup (rw,relatime,blkio)
cgroup on /cgroup/cpu type cgroup (rw,relatime,cpu)
cgroup on /cgroup/cpuacct type cgroup (rw,relatime,cpuacct)
cgroup on /cgroup/cpuset type cgroup (rw,relatime,cpuset)
cgroup on /cgroup/devices type cgroup (rw,relatime,devices)
cgroup on /cgroup/freezer type cgroup (rw,relatime,freezer)
cgroup on /cgroup/hugetlb type cgroup (rw,relatime,hugetlb)
cgroup on /cgroup/memory type cgroup (rw,relatime,memory)
cgroup on /cgroup/perf_event type cgroup (rw,relatime,perf_event)

Comment (pvam) [inline rubix_admin/installer.py]: Can go in /mnt/some-folder i think. /var is also a symlink to /mnt

**Comment by vrajat** [comment]
Which one of these is the largest device ? My guess is /emr.

/dev/xvdb1 on /emr type xfs (rw,relatime,attr2,inode64,noquota)
/dev/xvdb2 on /mnt type xfs (rw,relatime,attr2,inode64,noquota)
/dev/xvdc on /mnt1 type xfs (rw,relatime,attr2,inode64,noquota)
/dev/xvdd on /mnt2 type xfs (rw,relatime,attr2,inode64,noquota)
/dev/xvde on /mnt3 type xfs (rw,relatime,attr2,inode64,noquota)

**Comment by vrajat** [comment]
Spoke offline. Lets choose /var/lib/rubix.

**Pull request merged**